### PR TITLE
♻️ Move generic hashing and storage utilities from `lamindb` into `lamindb-setup`

### PR DESCRIPTION
**Pull request by bpenteado**
- https://github.com/laminlabs/lamindb/pull/1476
- https://www.notion.so/laminlabs/Monitor-hosted-storage-usage-enable-safe-deletes-4afbd336989c4f578fc8fab2194328eb

### CODE REVIEW

**Comment by falexwolf** [comment]
Could you cross-link this with the work package?

**Comment by bpenteado** [comment]
> Could you cross-link this with the work package?

Done!

File: lamindb_setup/core/upath.py [hunk 1]
```diff
@@ -484,3 +485,65 @@ def create_path(path: UPath, access_token: Optional[str] = None) -> UPath:
         secret=credentials["secret"],
         token=credentials["token"],
     )
+
+
+def get_stat_file_cloud(stat: Dict) -> Tuple[int, str, str]:
```
Comment (falexwolf) [inline lamindb_setup/core/upath.py:490]: We need tests for this, @bpenteado!

In lamindb, this was covered, now, I fear, it's uncovered by CI.

My guess is that there were no fine-grained unit tests in lamindb for this, but coverage came through more high-level functionality.

Given this is in lamindb-setup now and the interfaces are used by laminhub and lamindb, we need a more fine-grained test that reflects needs in both downstream repos (they should be the same if we design the code well).

Comment (falexwolf) [reply lamindb_setup/core/upath.py:490]: Here is an issue for this: https://github.com/laminlabs/lamindb-setup/issues/665

File: tests/test_hashing.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import base64
```
Comment (falexwolf) [inline tests/test_hashing.py:1]: I fear these tests don't executed - see the noxfile. I'll take care of this.

Comment (falexwolf) [reply tests/test_hashing.py:1]: Fixed here: https://github.com/laminlabs/lamindb-setup/pull/666

**Pull request merged**